types/node/test/events_generic.ts

```diff
@@ -0,0 +1,190 @@
+import * as events from "node:events";
+
+interface T {
+    event1: [string, number];
+    event2: [boolean];
+    event3: [];
+    event4: string[];
+    event5: unknown[];
+}
+
+const emitter = new events.EventEmitter<T>();
+declare const listener1: (...args: T["event1"]) => void;
+declare const listener2: (...args: T["event2"]) => void;
+declare const listener3: (...args: T["event3"]) => void;
+declare const listener4: (...args: T["event4"]) => void;
+declare const listener5: (...args: T["event5"]) => void;
+declare const event1: "event1";
+declare const event2: "event2";
+declare const event3: "event3";
+declare const event4: "event4";
+declare const event5: "event5";
+
+{
+    let result: events.EventEmitter<T>;
+
+    result = emitter.addListener(event1, listener1);
+    result = emitter.on(event1, listener1);
+    result = emitter.once(event1, listener1);
+    result = emitter.prependListener(event1, listener1);
+    result = emitter.prependOnceListener(event1, listener1);
+    result = emitter.removeListener(event1, listener1);
+    result = emitter.off(event1, listener1);
+    result = emitter.removeAllListeners();
+    result = emitter.removeAllListeners(event1);
+    result = emitter.setMaxListeners(42);
+
```

Comment:
Consider testing with less strict listener types, such as:

```ts
emitter.addListener(event1, () => undefined);
emitter.addListener(event1, (a: string) => undefined);
emitter.addListener(event1, (a: string, b?: number) => undefined);
emitter.addListener(event1, (a: string, b: number | boolean): number => 1);
```

Reply:
Added (verbatim, plus one more that's intentionally erroneous) in 50dca4c91e05506dc215e9209691d144119bf95f.